### PR TITLE
[RISCV] Sink ordered reduction check into FAdd. NFC

### PR DESCRIPTION
**Pull request by ElvisWang123**

### CODE REVIEW

**Comment by preames** [comment]
What is the point of this change?  Only floating point reductions should return true for requiresOrderedReduction?

**Comment by ElvisWang123** [comment]
> What is the point of this change? Only floating point reductions should return true for requiresOrderedReduction?

Without the check  querying  the `getArithmeticReductionCost()` by `reduce.add` `reduce.and` ... integer  reductions  without  `reassoc` flag will fall into it. And it will return the cost of the floating point ordered reduction.

I found this issue in the #114184. When we query the cost model and always pass the FMF flags from the IntrinsicCostAttributes, it always return the cost of FP ordered reduction.

I thought it  is more user friendly to check if it is floating point type and user can call `getArithmeticReductionCost()` without   dropping `FMF` flags when it is integer type.

**Comment by preames** [comment]
> Without the check querying the `getArithmeticReductionCost()` by `reduce.add` `reduce.and` ... integer reductions without `reassoc` flag will fall into it. And it will return the cost of the floating point ordered reduction.

Which caller is this coming from?  getTypeBasedIntrinsicInstrCost (the one used by CostAnalysis printer pass), appears to pass nullopt for everything except fadd/fmul.

**Comment by preames** [comment]
Mostly summarizing my investigation for future reference.

The way this works for the current reduce.* intrinsics is that BasicTTI dispatches these to getArithmeticReductionCost and getMinMaxReductionCost respective.  BasicTTI distinguishes fadd/fmul from the other arithmetic, and specifically passes None for the FMF on non-FP arithmetics.  It does pass the FMF to the MinMax variant, but since there are no ordered min/max variants, that doesn't apply.

File: llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp
Comment (preames) [inline llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp:1555]: I think a cleaner way to express this change is to simply sink the check for requiresOrderedReduction under the FADD case in the switch below.  That's the only supported reduction kind with an ordered variant.

**Comment by ElvisWang123** [comment]
I am worried about  transformations using `getArithmeticReductionCost()` directly  to query the instruction cost and not dropping the FMF flags with integer type.
After investigation, all query will not return wrong instruction cost currently. 

I am fine to close this PR.

**Review by preames — APPROVED** [review]
LGTM

**Review by lukel97 — APPROVED** [review]
LGTM, I'll rework #114250 after this lands

**Pull request merged**
